### PR TITLE
docs: fetch external sources

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

making docs as standalone resources without needing to pull S3 assets
pending on https://github.com/Lightning-AI/utilities/pull/142

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @tchaton @carmocca @justusschock @awaelchli

### CODE REVIEW

**Comment by Borda** [comment]
okk, it seems that Sphinx does not include resources linked only in HTML section

```
.. raw:: html

    <video width="50%" max-width="400px" controls
    poster="https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/trainer_flags/thumb/check_val_every_n_epoch.jpg"
    src="https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/trainer_flags/check_val_every_n_epoch.mp4"></video>

File: requirements/docs.txt [hunk 1]
```diff
@@ -13,4 +13,8 @@ sphinx-copybutton >=0.3, <=0.5.2
 sphinx-multiproject
 sphinx-toolbox ==3.4.0
 sphinx-rtd-dark-mode
+sphinxcontrib-video ==0.2.0
 jinja2 >=3.0.0,<3.2.0
+
+# todo: replace with released version, likely 0.9
+lightning-utilities @ https://github.com/Lightning-AI/utilities/archive/refs/heads/docs/extracor.zip
```
Comment (Borda) [inline requirements/docs.txt:20]:
```suggestion
lightning-utilities @ https://github.com/Lightning-AI/utilities/archive/refs/heads/main.zip
```

**Review by rlizzo — APPROVED** [review]

**Review by lantiga — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by edenlightning — APPROVED** [review]

File: requirements/docs.txt [hunk 2]
```diff
@@ -13,4 +13,8 @@ sphinx-copybutton >=0.3, <=0.5.2
 sphinx-multiproject
 sphinx-toolbox ==3.4.0
 sphinx-rtd-dark-mode
+sphinxcontrib-video ==0.2.0
 jinja2 >=3.0.0,<3.2.0
+
+# todo: replace with released version, likely 0.9
+lightning-utilities @ https://github.com/Lightning-AI/utilities/archive/refs/heads/main.zip
```
Comment (carmocca) [inline requirements/docs.txt:20]: Shouldn't this set 
```suggestion
lightning-utilities >=0.9.0
```

As in the other requirement files?

**Review by carmocca — APPROVED** [review]
There's no docs preview so I don't really know what I'm approving :crossed_fingers:

File: docs/source-fabric/fundamentals/launch.rst [hunk 3]
```diff
@@ -15,11 +15,11 @@ To run your code distributed across many devices and many machines, you need to
 Simple Launch
 *************
 
-.. raw:: html
```
Comment (awaelchli) [inline docs/source-fabric/fundamentals/launch.rst:18]: This changes the format drastically. The video is no longer centered and much smaller, and controls are visible. See the original page and compare the two.

Comment (Borda) [reply docs/source-fabric/fundamentals/launch.rst:18]: ok, there is not much we can do about the center unless define in the template's CSS class... this the generated HTML source:
```html
<video autoplay="True" controls="True" loop="True" muted="True" preload="auto" width="500"><source src="[../_static/fetched-s3-assets/launch.mp4](https://github.com/Lightning-AI/lightning/pull/_static/fetched-s3-assets/launch.mp4)" type="video/mp4"></video>
```

Comment (Borda) [reply docs/source-fabric/fundamentals/launch.rst:18]: so it seems your original HTML was wrong as it ignored `max-width="500px"`
about controls, seems that `.. video::` has it on by default and disable need to be explicitly set by `:nocontrols:`

Comment (Borda) [reply docs/source-fabric/fundamentals/launch.rst:18]: tbh, I would always rather show the controls so the user can pause or better navigate it...

File: docs/source-fabric/fundamentals/launch.rst [hunk 4]
```diff
@@ -15,11 +15,11 @@ To run your code distributed across many devices and many machines, you need to
...
-
-    <center>
-        <video width="100%" max-width="500px" src="https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/launch.mp4", type="video/mp4" autoplay loop muted></video>
-    </center>
+.. video:: https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/launch.mp4
+    :width: 500
```
Comment (Borda) [inline docs/source-fabric/fundamentals/launch.rst:19]:
```suggestion
    :width: 800
```
compensation for wrong HTML

File: docs/source-fabric/fundamentals/launch.rst [hunk 5]
```diff
@@ -56,11 +56,11 @@ You can learn more about it :ref:`here <Fabric in Notebooks>`.
 Launch with the CLI
 *******************
 
-.. raw:: html
-
-    <center>
-        <video width="100%" max-width="500px" src="https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/launch-cli.mp4", type="video/mp4" autoplay loop muted></video>
-    </center>
+.. video:: https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/launch-cli.mp4
+    :width: 500
```
Comment (Borda) [inline docs/source-fabric/fundamentals/launch.rst:60]:
```suggestion
    :width: 800
```
compensation for wrong HTML

File: docs/source-fabric/fundamentals/precision.rst [hunk 6]
```diff
@@ -2,11 +2,11 @@
 Save memory with mixed precision
 ################################
 
-.. raw:: html
-
-    <center>
-        <video width="100%" max-width="500px" src="https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/precision.mp4", type="video/mp4" autoplay loop muted></video>
-    </center>
+.. video:: https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/precision.mp4
+    :width: 500
```
Comment (Borda) [inline docs/source-fabric/fundamentals/precision.rst:6]:
```suggestion
    :width: 800
```
compensation for wrong HTML

File: docs/source-fabric/fundamentals/accelerators.rst [hunk 7]
```diff
@@ -3,11 +3,11 @@ Accelerate your code with Fabric
 ################################
 
 
-.. raw:: html
-
-    <center>
-        <video width="100%" max-width="500px" src="https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/accelerators.mp4", type="video/mp4" autoplay loop muted></video>
-    </center>
+.. video:: https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric/animations/accelerators.mp4
+    :width: 500
```
Comment (Borda) [inline docs/source-fabric/fundamentals/accelerators.rst:7]:
```suggestion
    :width: 800
```
compensation for wrong HTML

File: src/lightning/fabric/loggers/csv_logs.py [hunk 8]
```diff
@@ -125,11 +125,13 @@ def experiment(self) -> "_ExperimentWriter":
         return self._experiment
 
     @rank_zero_only
-    def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
+    def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:  # type: ignore[override]
```
Comment (carmocca) [inline src/lightning/fabric/loggers/csv_logs.py:128]: Why are we adding all these typing changes with this PR? How does this relate to the docs changes?

Comment (Borda) [reply src/lightning/fabric/loggers/csv_logs.py:128]: `mypy` started to complain after using the latest Utils, and without fix I can't merge it so I could also bump it else were :)

**Review by awaelchli — APPROVED** [review]

**Pull request merged**